p2p/syncer.go

```diff
@@ -239,11 +239,13 @@ type Syncer struct {
 	cm     *chain.Manager
 	tp     *txpool.Pool
 	header gateway.Header
+	store  SyncerStore
 
-	cond  sync.Cond
-	mu    sync.Mutex
-	peers map[gateway.Header]*gateway.Session
-	err   error
+	cond     sync.Cond
+	peerCond sync.Cond
+	mu       sync.Mutex
+	peers    map[gateway.Header]*gateway.Session
+	err      error
```

Comment:
let's stick with one cond for now. Wakeups are going to be fairly uncommon, and spurious wakeups are cheap -- the goroutine just checks a few variables and goes back to sleep.